include/seqan3/io/sam_file/record.hpp

```diff
@@ -123,33 +123,34 @@ class sam_record : public record<field_types, field_ids>
         return get_impl(field_constant<seqan3::field::qual>{}, static_cast<tuple_base_t const &>(*this));
     }
 
-    /*!\brief Sequence (seqan3::sam_record::sequence) relative start position (0-based), unsigned value.
-     *        (SAM Column ID: POS)
-     * \returns seqan3::sam_file_input::offset_type per default
+    /*!\brief [DEPRECATED] Sequence (seqan3::sam_record::sequence) relative start position (0-based), unsigned value.
+     * \returns 0
      *
-     * \details
+     * \deprecated This member is deprecated please access the `seqan3::sam_record::cigar()` information directly and
+     *             check the value for soft clipping (S) at the front of the CIGAR string. It is synonym to the
+     *             offset. If not soft clipping is present at the front, the offset is zero.
      *
      * The position is the length of the soft-clipping at the start of the seqan3::sam_record::cigar_sequence if a
      * soft-clipping is present and `0` otherwise.
      */
-    decltype(auto) sequence_position() &&
+    SEQAN3_DEPRECATED_340 decltype(auto) sequence_position() &&
     {
-        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<tuple_base_t &&>(*this));
+        return static_cast<int32_t>(0);
```

Comment:
```suggestion
        return format_sam_base::soft_clipping_at_front(cigar()); // Won't work as the base is protected
```

Wouldn't we want something like this? Such that it is deprecated, but still does the correct thing.
Thinking about it, `alignment()` should probably throw instead of just doing nothing...though you cannot get to call `alignment()` because it will static assert. 

I guess it's the same here :D
One could argue that it might help to provide the `soft_clipping_at_front` function. It does seem like a common use case.

Reply:
> Wouldn't we want something like this? Such that it is deprecated, but still does the correct thing.

That was also my first impulse but it is not guaranteed that `cigar` is available. Maybe we should not return anything to mimic the alingment removal and avoid unexpected semantic issues.

> Thinking about it, alignment() should probably throw instead of just doing nothing...though you cannot get to call alignment() because it will static assert.

Yes, you would never get to the exception because you would need to ignore the deprecation warning and then just use `rec.alignment()` without catching the result. When trying to catch it, you will get the error that alignment is void.

> One could argue that it might help to provide the soft_clipping_at_front function. It does seem like a common use case. Especially since there seems to be a pitfall where there might be hard clipping in the first position and the soft clipping in the second.

Yes. But this would be part of several utility functions for a cigar vector.
We need to discuss them separately and should not implement them without a use case.

Utility that's be nice:
* CIGAR String <-> CIGAR Vector transformation
* Get Hard/Softclipping at front or back
* Get Length of the reference sequence region
* Get Length of the query sequence region

Reply:
So in summary, I would remove the `return` completely, analog to `alignment()`.

Reply:
I agree, should just be
```cpp
SEQAN3_DEPRECATED_340 decltype(auto) sequence_position() &&
{}
```
etc.